diff --git a/driver/postgres/PostgresDriver.js b/driver/postgres/PostgresDriver.js
index 278f29c1f3deec4939bb4ed90e6edae167f704e0..9a84c3098dda915d6c33e24d925a8fa09af9095e 100644
--- a/driver/postgres/PostgresDriver.js
+++ b/driver/postgres/PostgresDriver.js
@@ -785,10 +785,10 @@ class PostgresDriver {
             const tableColumnDefault = typeof tableColumn.default === "string"
                 ? JSON.parse(tableColumn.default.substring(1, tableColumn.default.length - 1))
                 : tableColumn.default;
-            return OrmUtils_1.OrmUtils.deepCompare(columnMetadata.default, tableColumnDefault);
+            return OrmUtils_1.OrmUtils.deepCompare(columnMetadata.default, tableColumnDefault ?? null);
         }
         const columnDefault = this.lowerDefaultValueIfNecessary(this.normalizeDefault(columnMetadata));
-        return columnDefault === tableColumn.default;
+        return columnDefault === tableColumn.default || columnDefault === undefined && tableColumn.default.toLowerCase() === 'null';
     }
     /**
      * Normalizes "isUnique" value of the column.
