diff --git a/dist/node/chunks/dep-DrOo5SEf.js b/dist/node/chunks/dep-DrOo5SEf.js
index 329e68bd27e55a56d815fa6b4de2d615a8c2b343..9d9f58e90ae836f80063b698e307fec436e53e07 100644
--- a/dist/node/chunks/dep-DrOo5SEf.js
+++ b/dist/node/chunks/dep-DrOo5SEf.js
@@ -45971,7 +45971,7 @@ function importAnalysisPlugin(config) {
       let isPartiallySelfAccepting = false;
       const importedBindings = enablePartialAccept ? /* @__PURE__ */ new Map() : null;
       const toAbsoluteUrl = (url) => path$d.posix.resolve(path$d.posix.dirname(importerModule.url), url);
-      const normalizeUrl = async (url, pos, forceSkipImportAnalysis = false) => {
+      const normalizeUrl = async (url, pos, forceSkipImportAnalysis = false, stripBase2 = false) => {
         url = stripBase(url, base);
         let importerFile = importer;
         if (depsOptimizer && moduleListContains(depsOptimizer.options.exclude, url)) {
@@ -46031,7 +46031,7 @@ function importAnalysisPlugin(config) {
           e.pos = pos;
           throw e;
         }
-        if (!ssr) url = joinUrlSegments(base, url);
+        if (!ssr && !stripBase2) url = joinUrlSegments(base, url);
         return [url, resolved.id];
       };
       const orderedImportedUrls = new Array(imports.length);
@@ -46288,7 +46288,7 @@ See ${colors$1.blue(
         const pluginImports = this._addedImports;
         if (pluginImports) {
           (await Promise.all(
-            [...pluginImports].map((id) => normalizeUrl(id, 0, true))
+            [...pluginImports].map((id) => normalizeUrl(id, 0, true, true))
           )).forEach(([url]) => importedUrls.add(url));
         }
         if (ssr && importerModule.isSelfAccepting) {
